python/mxnet/initializer.py

```diff
@@ -268,6 +268,10 @@ def _init_default(self, name, _):
             '"weight", "bias", "gamma" (1.0), and "beta" (0.0).' \
             'Please use mx.sym.Variable(init=mx.init.*) to set initialization pattern' % name)
 
+    def __eq__(self, other):
+        if not isinstance(other, Initializer):
+            return NotImplemented
+        return type(self) == type(other) and self._kwargs == other._kwargs
```

Comment:
does the second statement behave correctly if it involves constant initializer (w/ numpy array, NDArray, and mxnet numpy array)?

Reply:
We can say it behaves correctly, if we expect `Constant(10.0) != Constant(np.array([10.0])`.

Reply:
With this PR at least `Constant(10.0) == Constant(10.0)`, which was not true before...